### ISSUE TITLE
error: use of undeclared identifier 'MAP_POPULATE'

### ISSUE THREAD

**Issue by liu-jc** [rdspring1/MISSION#1]
I am interested in your paper. However, when I tried to run this code, I got some problems. When I type make all, I got following outputs: 

```
rm -rf MurmurHash.o
rm -rf fast_parser.o
rm -rf util.o
rm -rf mission_logistic
rm -rf fine_mission_softmax
rm -rf coarse_mission_softmax
rm -rf softmax
g++ -Wall --std=c++14 -O3 -Iinclude/ -o fast_parser.o -c fast_parser.cpp
fast_parser.cpp:31:66: error: use of undeclared identifier 'MAP_POPULATE'
                addr = mmap(NULL, pg_size, PROT_READ, MAP_FILE | MAP_PRIVATE | MAP_POPULATE, fd, offset);
                                                                               ^
1 error generated.
make: *** [fast_parser] Error 1
```

How can I fix this? I think MAP_POPULATE should be given in mman.h ?

**Comment by rdspring1** [comment]
MAP_POPULATE is available only on Linux and only since version 2.5.46 (since version 2.6.23 for private mappings)

**Comment by rdspring1** [comment]
What version of Linux are you using?
